sklearn/feature_extraction/image.py

```diff
@@ -139,6 +138,14 @@ def _to_graph(
     return return_as(graph)
 
 
+@validate_params(
+    {
+        "img": ["array-like"],  # Could it also be "sparse matrix"?
+        "mask": [None, "array-like"],
```

Comment:
Since we call `mask.astype(...)` or `mask.ravel()`, we expect `mask` to be a `ndarray` as stipulated by the documentation:

```suggestion
        "mask": [None, np.ndarray],
```

Reply:
Thanks for the catch. I think I will write a test for the underlying function because it seems more flexible. It confusingly had a redundant `asarray` call  after the `.astype()` for the `mask`—not needed since python 2.7—which I removed since it was basically doing validation.